backend/src/auth/services/auth.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from "@nestjs/common";
+import { JwtService } from "@nestjs/jwt";
+import * as bcrypt from "bcrypt";
+import { UsersService } from "users/services/users.service";
+import { UserCredentialsDto } from "users/model/dto/userCredentials.dto";
+import { User } from "users/model/users.entity";
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UsersService,
+    private readonly jwtService: JwtService
+  ) {
+  }
+
+  async login({ username }: UserCredentialsDto) {
+    username = username.replace(/\s/g, "").toLowerCase();
+
+    const { id } = await this.userService.findOneByUsername(username);
+
+    return { accessToken: await this.generateJwt({ id, username }) };
+  }
+
+  async validateUser(userCredentialsDto: UserCredentialsDto) {
+    const { password } = await this.userService.passwordSelect(userCredentialsDto.username);
+
+    const isPasswordValid = await this.comparePasswords(
+      userCredentialsDto.password,
+      password
+    );
+
+    if (isPasswordValid)
+      return userCredentialsDto;
+  }
+
+  async register(loginRegisterUserDto: UserCredentialsDto) {
+    loginRegisterUserDto.username = loginRegisterUserDto.username
+      .replace(/\s/g, "").toLowerCase();
+
+    const hashedPassword = await bcrypt.hash(loginRegisterUserDto.password, 15);
+
+    const user = await this.userService.create({
+      ...loginRegisterUserDto,
+      password: hashedPassword
+    });
+
+    const { password, ...rest } = user;
+
+    return { accessToken: await this.generateJwt(rest) };
+  }
+
+  private async generateJwt(user: User) {
```

Comment:
Make this method return { access_token: string }

Reply:
That method do it currently

Reply:
Done